### PR TITLE
Updating StFcsTriggerSimMake to include StMuFcsCollection hit information for trigger reconstruction in simulation.

### PR DESCRIPTION
**Pull request by mrosales2016**
The updates include StMuFcsCollection so that users of this maker may be able to use either StEvent fcsHit information or MuDst fcsHit information for trigger reconstruction in simulation.

### CODE REVIEW

**Review by zlchang — APPROVED** [review]

File: StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx [hunk 1]
```diff
@@ -279,43 +305,83 @@ int StFcsTriggerSimMaker::Make(){
...
+                }
+                  }
+                  mTrgSim->fill_event(ehp,ns,dep,ch,data,8) ;
+                }else{
+                  data[1] = hit->adc(0)-1;  //removing 1 to add at tb6
+                  data[6] = 1;              //add this so tb6>tb7
+                  mTrgSim->fill_event(ehp,ns,dep,ch,data,8) ;
+                }
+                if(mFile) fprintf(mFile,"%2d %2d %2d %2d %5d\n",ns,ehp,dep,ch,hit->adc(0));
+                n++;
```
Comment (veprbl) [inline StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx]: Would you mind moving the common code into a separate function?

Comment (mrosales2016) [reply StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx]: Sure thing. I will go ahead and make the changes thank you for the suggestion.

Comment (mrosales2016) [reply StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx]: I just added my modification after your suggestions to my local fork of the star-sw page. If i go to the initial pull request that I made I can see the commit with the new changes there but I'm not sure if those would be come the new ones here. Can you please tell me if you see the new commits and whether the updates follow your suggestions.

File: StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx [hunk 2]
```diff
@@ -656,3 +689,56 @@ void StFcsTriggerSimMaker::readThresholdFile(){
 void StFcsTriggerSimMaker::readThresholdDb(){
   //to be implemented before run22 online DB moves to offline  
 }
+
+void StFcsTriggerSimMaker::feedADC(StFcsHit* hit, int ns, int ehp, uint16_t data_array[]){
```
Comment (veprbl) [inline StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx]:
```suggestion
template<typename T>
void StFcsTriggerSimMaker::feedADC(T* hit, int ns, int ehp, uint16_t data_array[]){
```

File: StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx [hunk 3]
```diff
@@ -656,3 +689,56 @@ void StFcsTriggerSimMaker::readThresholdFile(){
...
+      mTrgSim->fill_event(ehp,ns,dep,ch,data_array,8) ;
+    }else{
+      data_array[1] = hit->adc(0)-1;  //removing 1 to add at tb6
+      data_array[6] = 1;              //add this so tb6>tb7
+      mTrgSim->fill_event(ehp,ns,dep,ch,data_array,8) ;
+    }
+    if(mFile) fprintf(mFile,"%2d %2d %2d %2d %5d\n",ns,ehp,dep,ch,hit->adc(0));
+ 
+}
+
```
Comment (veprbl) [inline StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx]:
```suggestion
```

File: StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.h [hunk 4]
```diff
@@ -42,13 +46,19 @@ class StFcsTriggerSimMaker : public StMaker{
     void setEtGain(float v) {mEtFactor=v;} 
 
     fcs_trg_base* getTriggerEmu() {return mTrgSim;}
+    
+    void feedADC(StFcsHit* hit, int ns, int ehp, uint16_t data_array[]);
+    void feedADC(StMuFcsHit* hit, int ns, int ehp, uint16_t data_array[]);
```
Comment (veprbl) [inline StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.h]:
```suggestion
    template<typename T>
    void feedADC(T* hit, int ns, int ehp, uint16_t data_array[]);
```

**Review by veprbl — COMMENTED** [review]
Looks like you can template this code.

File: StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx (hunk 1 above)
Comment (veprbl) [reply StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.cxx]: I will mark this one as resolved.

File: StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.h (hunk 4 above)
Comment (mrosales2016) [reply StRoot/StSpinPool/StFcsTriggerSimMaker/StFcsTriggerSimMaker.h]: Great suggestion. Just made the change let me know if that checkout

**Review by veprbl — APPROVED** [review]
Well done

**Review by akioogawa — APPROVED** [review]
LGTM

**Pull request merged**